eslint.config.js

```diff
@@ -56,7 +56,13 @@ module.exports = [
               name: 'styled-components',
               message: 'Please import from styled-components/macro.',
             },
+            {
+              name: 'react-router-dom',
+              importNames: ['useNavigate'],
+              message: "Please import useNavigate from our own common package instead: 'common/hooks/useNavigate'",
```

Comment:
What about the `explorer` app?

I see `useNavigate` being used in a few places there
![image](https://github.com/cowprotocol/cowswap/assets/43217/d9b98e11-1b22-4c09-868d-431784f97b07), 

This hook is only for `cowswap-frontend` app, while the rule applies to the whole app.

We should either have the eslint rule apply only to cowswap or apply the hook everywhere.

Reply:
Okay, I pushed a fix for this. I was doing the reverse, but no need to do it because this is not exported from the common-hooks module so there's no way for the explorer to import it (you only get suggested the react router one)

This is the test of the new rules. I also simplified the glob for selecting the files we had:

<img width="1098" alt="Screenshot at May 23 18-41-38" src="https://github.com/cowprotocol/cowswap/assets/2352112/2ff52b0d-cd53-4b99-8b2d-602e037feabc">

Reply:
BTW, refactored a bit the common-hooks module so it can be safely used in other apps https://github.com/cowprotocol/cowswap/pull/4478

Will hit develop once the arb1 PR is merged